components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1681,15 +1682,16 @@ public void addIdP(IdentityProvider identityProvider, int tenantId)
 
             prepStmt.setString(4, identityProvider.getHomeRealmId());
 
-            if (StringUtils.isNotBlank(identityProvider.getCertificate())) {
+            if (ArrayUtils.isNotEmpty(identityProvider.getCertificateInfoArray())) {
                 try {
-                    IdentityApplicationManagementUtil.getCertData(identityProvider.getCertificate());
+                    IdentityApplicationManagementUtil.getCertDataArray(identityProvider.getCertificateInfoArray());
```

Comment:
Why is this method called? Result of the method call is not used.

Reply:
The reason is to check whether certificate decoding and certificate generation fails or not.

Reply:
Better to add a comment to mention the reason